dtfs-central-api/api-tests/v1/portal-facility-amendments/amendments-on-deal-get.api-test.ts

```diff
@@ -0,0 +1,148 @@
+import { Response } from 'supertest';
+import { HttpStatusCode } from 'axios';
+import { API_ERROR_CODE, MONGO_DB_COLLECTIONS, PORTAL_AMENDMENT_STATUS, PortalAmendmentStatus, PortalFacilityAmendment, TfmFacility } from '@ukef/dtfs2-common';
+import { aPortalFacilityAmendment } from '@ukef/dtfs2-common/mock-data-backend';
+import { ObjectId } from 'mongodb';
+import wipeDB from '../../wipeDB';
+import { testApi } from '../../test-api';
+import { mongoDbClient as db } from '../../../src/drivers/db-client';
+import { aCompletedTfmFacilityAmendment, aTfmFacility, aTfmFacilityAmendment } from '../../../test-helpers';
+
+interface PortalAmendmentsResponse extends Response {
+  body: PortalFacilityAmendment[];
+}
+
+const { DRAFT, ACKNOWLEDGED, READY_FOR_CHECKERS_APPROVAL, FURTHER_MAKERS_INPUT_REQUIRED } = PORTAL_AMENDMENT_STATUS;
+
+const generateUrl = ({ dealId, statuses }: { dealId: string; statuses?: PortalAmendmentStatus[] }): string => {
+  const statusFilterQuery = statuses ? `?statuses=${statuses.map((item) => item.replace(/ /g, '%20')).join(',')}` : '';
```

Comment:
I think using `encodeURI` would be more robust (e.g. handles `/` or non-ascii in the status)

```suggestion
  const statusFilterQuery = statuses ? `?statuses=${statuses.map((item) => encodeUri(item)).join(',')}` : '';
```

Reply:
Beautiful, did not know that was a thing✨